text-document.rkt

```diff
@@ -619,7 +618,7 @@
   (cond
     [(not (number? desired-spaces)) #f]
     [(= current-spaces desired-spaces) #f]
-    [(= line-length 0) #f]
+    [(and (not (eq? kind 'on-type)) (= line-length 0)) #f]
```

Comment:
If we only need `'on-type`, maybe we can make it a flag for now? e.g.
```racket
(define (xxx #:on-type? [on-type? #f])
  ...)
```

Reply:
Ah sure, I'm happy to use a flag instead for this case. I'll wait a bit to see if there are any other comments, and then I'll make this change.

Reply:
I agree with dannypsnl's comment.